alicloud/service_alicloud_cen.go

```diff
@@ -5,14 +5,19 @@ import (
 	"strings"
 	"time"
 
+	"github.com/alibaba/terraform-provider/alicloud/aliyunclient"
 	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
 	"github.com/aliyun/alibaba-cloud-sdk-go/services/cbn"
 )
 
+type CenService struct {
+	client *aliyunclient.AliyunClient
```

Comment:
I think line 14 just  using `*aliyunclient.AliyunClient` is ok.

Reply:
I disagree, the semantic changes when you remove "client": in the current implementation a service **uses** the client, but in what you propose the service **extends** the client.

Although it is slightly more verbose to write "myService.client.SomeFunction" instead of "myService.SomeFunction", it is easier to read because the reader can immediately understand that the function come from the client, not the service.

It's like the [Composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance) principle.